warehouse/oidc/forms/activestate.py

```diff
@@ -0,0 +1,245 @@
+)
+TIMEOUT = 5
+
+
+class UserResponse(TypedDict):
+    user_id: str
+
+
+class ActiveStatePublisherBase(forms.Form):
+    __params__ = ["organization", "project", "actor"]
+
+    organization = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=_("Specify ActiveState organization name"),
+            ),
+            wtforms.validators.Regexp(
+                _VALID_ORG_URL_NAME,
+                message=_("Invalid ActiveState organization name"),
+            ),
+        ]
+    )
+
+    project = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=_("Specify ActiveState project name")
+            ),
+            wtforms.validators.Regexp(
+                _VALID_PROJECT_NAME,
+                message=_("Invalid ActiveState project name"),
+            ),
+        ]
+    )
+
+    actor = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=("Specify the ActiveState actor username")
+            ),
+            wtforms.validators.Regexp(
+                _VALID_ORG_URL_NAME,
+                message=("Invalid ActiveState username"),
+            ),
+        ]
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _lookup_organization(self, org_url_name: str) -> None:
+        """Make gql API call to the ActiveState API to check if the organization
+        exists"""
+        try:
+            response = requests.post(
+                f"{ACTIVESTATE_GRAPHQL_API_URL}",
+                json={
+                    "query": GRAPHQL_GET_ORGANIZATION,
+                    "variables": {"orgname": org_url_name},
+                },
+                timeout=TIMEOUT,
+            )
+            response.raise_for_status()
+        except requests.HTTPError:
+            sentry_sdk.capture_message(
+                f"Unexpected error from ActiveState organization lookup: {response.content!r}"  # noqa
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected error from ActiveState. Try again")
+            )
+        except requests.ConnectionError:
+            sentry_sdk.capture_message(
+                "Connection error from ActiveState organization lookup API (possibly offline)"  # noqa
+            )
+            raise wtforms.validators.ValidationError(
+                _(
+                    "Unexpected connection error from ActiveState. "
+                    "Try again in a few minutes"
+                )
+            )
+        except requests.Timeout:
+            sentry_sdk.capture_message(
+                "Timeout from ActiveState organization lookup API (possibly offline)"
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected timeout from ActiveState. Try again in a few minutes")
+            )
```

Comment:
Seems like we could probably condense these into:

```python
except (requests.HTTPError, requests.ConnectionError, request.Timeout) as e:
    sentry_sdk.capture_message(
        f"Unexpected error {e!r} from ActiveState organization lookup: {response.content!r}"
    )
```

Reply:
This is so nice.  Thanks!